File: infrastructure/src/main/java/com/study/platform/global/outbox/infrastructure/OutboxEventRepositoryAdapter.java
package com.study.platform.global.outbox.infrastructure;

import com.study.platform.global.outbox.model.OutboxEvent;
import com.study.platform.global.outbox.model.OutboxEventRepository;
import com.study.platform.global.outbox.model.OutboxEventStatus;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;

@Repository
@RequiredArgsConstructor
public class OutboxEventRepositoryAdapter implements OutboxEventRepository {

    private final OutboxEventJpaRepository outboxEventJpaRepository;

    @Override
    public OutboxEvent save(OutboxEvent event) {
        return outboxEventJpaRepository.save(event);
    }

    @Override
    public List<OutboxEvent> findAllByStatusAndCreatedAtBefore(OutboxEventStatus status, LocalDateTime createdAt) {
        return outboxEventJpaRepository.findAllByStatusAndCreatedAtBefore(status, createdAt);
    }

    @Override
    public int markSentById(Long id) {
        return outboxEventJpaRepository.markSentById(id);
    }

    @Override
    public void markFailedPermanently(Long id) {
        outboxEventJpaRepository.markFailedPermanently(id);
    }
}

File: infrastructure/src/main/java/com/study/platform/global/outbox/infrastructure/OutboxEventRepositoryAdapter.kt
package com.study.platform.global.outbox.infrastructure

import com.study.platform.global.outbox.model.OutboxEvent
import com.study.platform.global.outbox.model.OutboxEventRepository
import com.study.platform.global.outbox.model.OutboxEventStatus
import org.springframework.stereotype.Repository
import java.time.LocalDateTime

@Repository
class OutboxEventRepositoryAdapter(
    private val outboxEventJpaRepository: OutboxEventJpaRepository
) : OutboxEventRepository {

    override fun save(event: OutboxEvent): OutboxEvent =
        outboxEventJpaRepository.save(event)

    override fun findAllByStatusAndCreatedAtBefore(status: OutboxEventStatus, createdAt: LocalDateTime): List<OutboxEvent> =
        outboxEventJpaRepository.findAllByStatusAndCreatedAtBefore(status, createdAt)

    override fun markSentById(id: Long): Int =
        outboxEventJpaRepository.markSentById(id)

    override fun markFailedPermanently(id: Long) {
        outboxEventJpaRepository.markFailedPermanently(id)
    }
}
